multi-agent/adapters/outbound/mongo/builtin_resource_descriptor_repository.py

```
@@ -0,0 +1,186 @@
+        sort_order: str,
+    ) -> Tuple[List[Resource], int]:
+        """Paginated resources visible to *identity*, joined with built-in
+        descriptor metadata — see the port's docstring for the three
+        ``ownership`` modes.
+
+        Anchored on ``resources`` (unlike ``find_all_builtins``) since the
+        default/custom modes must include the identity's own non-built-in
+        resources, which have no descriptor to anchor on at all.
+        """
+        if limit is not None and limit < 0:
+            raise ValueError(f"limit must be non-negative, got {limit}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)